File: exercism/go/isogram/isogram.go
// Package isogram determinates if a word is an isogram
package isogram

import (
	"strings"
	"unicode"
)

// IsIsogram is a function to know if a word is an isogram
func IsIsogram(word string) (is bool) {
	wordLow := strings.ToLower(word)
	for _, letter := range wordLow {

		count := strings.Count(wordLow, string(letter))
		if count > 1 && unicode.IsLetter(letter) == true {
			return false
		}

	}

	return true
}

File: exercism/cpp/isogram/isogram.cpp
#include "isogram.h"

#include <map>
#include <cctype>

namespace isogram {

    bool is_isogram(const std::string word) {
        std::map<char, int> mp;

        for (char character : word){
            if (!std::isalpha(character)) {
                continue;
            }

            character = std::toupper(character);
            mp[character] += 1;

            if (mp[character] > 1) {
                return false;
            }

        }

        return true;
    }

}  // namespace isogram
